### PR TITLE
JSON.stringify(m.prop(x)) use x.toJSON method when possible

### PR DESCRIPTION
**Pull request by husio**
When JSON serializing `m.prop`, check if wrapped object does define `toJSON` method
and if so, return it's result.

I'm not sure how to properly test javascript code, so please check everything before.

### CODE REVIEW

File: test/mithril.prop.js [hunk 1]
```diff
@@ -42,6 +42,22 @@ describe("m.prop()", function () {
...
+		var prop = m.prop(new Date(999))
+		expect(JSON.stringify(prop)).to.equal('"1970-01-01T00:00:00.999Z"')
+	})
+
+	it("correctly stringifies object with toJSON method", function () {
+		function Thing(name) {
+			this.name = name
+		}
+		Thing.prototype.toJSON = function() {
+			return JSON.stringify({kind: 'Thing', name: this.name})
```
Comment (StephanHoyer) [inline test/mithril.prop.js]: I think it's easier to read if you just do this
```javascript
return {kind: 'Thing', name: this.name}´
```
than you can revert this quote spamming in the expect.

**Comment by isiahmeadows** [comment]
I thought JSON.stringify did that recursively, but I may be wrong.

**Comment by pygy** [comment]
@husio As @isiahmeadows pointed out in #1179,  a `null` check on `store` is required, otherwise you'll get errors when serializing props that hold `null` and `undefined`.

**Comment by isiahmeadows** [comment]
@husio Could you add tests for `store` being `null` (serializes to `null`) and `undefined` (disappears)?
